aiida_vasp/calcs/vasp.py

```diff
@@ -303,8 +303,10 @@ def write_poscar(self, dst):  # pylint: disable=unused-argument
         settings = self.inputs.get('settings')
         settings = settings.get_dict() if settings else {}
         poscar_precision = settings.get('poscar_precision', 10)
-        positions_dof = self.inputs.get('dynamics', {}).get('positions_dof')
-        if positions_dof is not None:
+        dynamics = self.inputs.get('dynamics')
+        if dynamics is not None:
+            dynamics = dynamics.get_dict() if dynamics else {}
+            positions_dof = dynamics.get('positions_dof')
```

Comment:
Here we set it if `None`. Should we not set it if it is `None`?

Reply:
I'm sorry I do not think I understand what you mean. The reason why I do it this way is because if `dynamics` is not found by `get` I think it will be defaulted as `None`. Which is the behaviour that should happen if you do not pass any `dynamics` options into your calculation.

Reply:
For `aiida-vasp` we try to stick to the `aiida-core`.

Reply:
Yes, here, if there are other tags under `dynamics` (possibly in the future), `positions_dof` would be set to `None` instead of not setting that at all.

Down the line we do `np.array(options['positions_dof'])` which would fail on `None` in the `transform_to_bool`. We could of course modify that, but maybe easier not to set it?

Reply:
Okay I understand what you mean now. Sure one could do it the simple way
```
        if dynamics is not None:
            dynamics = dynamics.get_dict()
            positions_dof = dynamics.get('positions_dof')
            if positions_dof is not None:
                options = {'positions_dof': positions_dof}
            else:
                options = None
        else:
            options = None
```

Not the most elegant though, I guess one can also do something like
```
options = None
if dynamics is not None:
     options = {}
    dynamics = dynamics.get_dict()
    positions_dof = dynamics.get('positions_dof')
    if positions is not None:
        options['positions_dof'] = positions_dof
```

Not sure if that is much better.

Reply:
Maybe this would also work:

```
options = None
if dynamics is not None:
    dynamics = dynamics.get_dict()
    positions_dof = dynamics.get('positions_dof')
    if positions is not None:
        options = {'positions_dof' = positions_dof}
```
I would maybe prefer the bottom one to avoid the two obvious duplicates.

Reply:
Sure I was just leaving the dictionary definition open in case in the future someone uses the `options` for other keys. But perhaps that is a problem to be dealt with at that time. I'll change it to this, run my tests and amend the PR.

Reply:
@espenfl Everything seems to be fine, I have amended the PR